internal/pkg/transformation/kubernetes.go

```diff
@@ -279,8 +299,41 @@ func (p *PodMapper) toDeviceToPod(
 
 	for _, pod := range devicePods.GetPodResources() {
 		for _, container := range pod.GetContainers() {
-			for _, device := range container.GetDevices() {
+			podInfo := PodInfo{
+				Name:      pod.GetName(),
+				Namespace: pod.GetNamespace(),
+				Container: container.GetName(),
+			}
+
+			if dynamicResources := container.GetDynamicResources(); len(dynamicResources) > 0 {
+				for _, dr := range dynamicResources {
+					for _, claimResource := range dr.GetClaimResources() {
+						draDriverName := claimResource.GetDriverName()
+						if draDriverName != DRAGPUDriverName {
+							continue
+						}
+						draPoolName := claimResource.GetPoolName()
+						draDeviceName := claimResource.GetDeviceName()
+						uuid := p.ResourceSliceManager.GetUUID(draPoolName, draDeviceName)
+						if uuid == "" {
+							slog.Info(fmt.Sprintf("No UUID for %s/%s", draPoolName, draDeviceName))
+							continue
+						}
 
+						drInfo := DynamicResourceInfo{
+							ClaimName:      dr.GetClaimName(),
+							ClaimNamespace: dr.GetClaimNamespace(),
+							DriverName:     draDriverName,
+							PoolName:       draPoolName,
+							DeviceName:     draDeviceName,
+						}
+						podInfo.DynamicResources = append(podInfo.DynamicResources, drInfo)
+						deviceToPodMap[uuid] = podInfo
```

Comment:
Further down in this function body I see that we totally have mixed key "types" in this map. Yes, the key is always a string. But I see, among others:
- `giIdentifier`
- `gpuUUID`
- `strings.Split(deviceID, gkeVirtualGPUDeviceIDSeparator)[0]`
- `deviceID`

That makes we wonder how important it is for us to store the resource slice UUID here as a key. Would  there be a problem using `pool+"/"+dev.Name` (what you've used in the DRAResourceSliceManager-maintained lookup table)?

I must say, I don't quite understand what the return value `deviceToPodMap` is really used for further down the line. All I noticed is that the keys are not all strictly UUIDs.